src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,137 @@
+import { useState, useEffect, useMemo } from 'react';
+import { useMatch, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import { getPeople } from '../api';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const match = useMatch('/people/:personId');
+  const selectedPersonId = match?.params.personId || '';
+  const [searchParams] = useSearchParams();
+  const currentGender = searchParams.get('sex');
+  const pickedCenturies = searchParams.getAll('centuries');
+  const sortName = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const query = searchParams.get('query');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople().then(result => {
+      setPeople(result);
+    })
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+  const containsIgnoreCase = (text: string, queryText: string) => (
```

Comment:
The `containsIgnoreCase` function should be moved outside of the component, as it does not depend on the component's state or props. This will prevent unnecessary re-creation of the function on each render.